app/src/main/java/com/ciandt/dragonfly/example/features/download/DownloadedFile.kt

```diff
@@ -0,0 +1,115 @@
+        get() {
+            when (status) {
+                DownloadManager.STATUS_FAILED -> return "STATUS_FAILED"
+                DownloadManager.STATUS_PAUSED -> return "STATUS_PAUSED"
+                DownloadManager.STATUS_PENDING -> return "STATUS_PENDING"
+                DownloadManager.STATUS_RUNNING -> return "STATUS_RUNNING"
+                DownloadManager.STATUS_SUCCESSFUL -> return "STATUS_SUCCESSFUL"
+                STATUS_CANCELLED -> return "STATUS_CANCELLED"
+                else -> return ""
+            }
+        }
+
+    val reasonText: String
+        get() {
+            when (reason) {
+                DownloadManager.ERROR_CANNOT_RESUME -> return "ERROR_CANNOT_RESUME"
+                DownloadManager.ERROR_DEVICE_NOT_FOUND -> return "ERROR_DEVICE_NOT_FOUND"
+                DownloadManager.ERROR_FILE_ALREADY_EXISTS -> return "ERROR_FILE_ALREADY_EXISTS"
+                DownloadManager.ERROR_FILE_ERROR -> return "ERROR_FILE_ERROR"
+                DownloadManager.ERROR_HTTP_DATA_ERROR -> return "ERROR_HTTP_DATA_ERROR"
+                DownloadManager.ERROR_INSUFFICIENT_SPACE -> return "ERROR_INSUFFICIENT_SPACE"
+                DownloadManager.ERROR_TOO_MANY_REDIRECTS -> return "ERROR_TOO_MANY_REDIRECTS"
+                DownloadManager.ERROR_UNHANDLED_HTTP_CODE -> return "ERROR_UNHANDLED_HTTP_CODE"
+                DownloadManager.ERROR_UNKNOWN -> return "ERROR_UNKNOWN"
+                DownloadManager.PAUSED_QUEUED_FOR_WIFI -> return "PAUSED_QUEUED_FOR_WIFI"
+                DownloadManager.PAUSED_UNKNOWN -> return "PAUSED_UNKNOWN"
+                DownloadManager.PAUSED_WAITING_FOR_NETWORK -> return "PAUSED_WAITING_FOR_NETWORK"
+                DownloadManager.PAUSED_WAITING_TO_RETRY -> return "PAUSED_WAITING_TO_RETRY"
+                else -> return ""
+            }
+        }
+
+    fun isPending(): Boolean = status == DownloadManager.STATUS_PENDING
+
+    fun isRunning(): Boolean = status == DownloadManager.STATUS_RUNNING
+
+    fun isPaused(): Boolean = status == DownloadManager.STATUS_PAUSED
+
+    fun isSuccessful(): Boolean = status == DownloadManager.STATUS_SUCCESSFUL
+
+    fun isFailed(): Boolean = status == DownloadManager.STATUS_FAILED
+
+    fun isCancelled(): Boolean = status == STATUS_CANCELLED
+
+    fun isFinished(): Boolean = isSuccessful() || isFailed() || isCancelled()
+
+    override fun toString(): String {
+        return "DownloadedFile(" + "\n" +
+                "\tid=$id, " + "\n" +
+                "\ttitle=$title, " + "\n" +
+                "\tstatus=$status, " + "\n" +
+                "\tstatusText=$statusText, " + "\n" +
+                "\treason=$reason, " + "\n" +
+                "\treasonText=$reasonText, " + "\n" +
+                "\tbytesTotal=$bytesTotal, " + "\n" +
+                "\tbytesDownloaded=$bytesDownloaded, " + "\n" +
+                "\tlastModifiedAt=$lastModifiedAt" + "\n" +
+                ")"
+    }
+
+    constructor(parcel: Parcel) : this(
```

Comment:
Maybe that's worth considering? “Reducing Parcelable boilerplate code using Kotlin” -
 https://medium.com/@BladeCoder/reducing-parcelable-boilerplate-code-using-kotlin-741c3124a49a